### ISSUE TITLE
Can't find declaration to go to when Ctrl-Clicking a bindgen ffi method from crate with build script

### ISSUE THREAD

**Issue by zumoshi** [intellij-rust/intellij-rust#6637]
## Environment

* **IntelliJ Rust plugin version:** 0.3.138.3572-203
* **Rust toolchain version:** 1.48.0 x86_64-pc-windows-gnu
* **IDE name and version:** CLion 2020.3.1 (CL-203.6682.181)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** old

## Problem description

Can't find declaration to go to when Ctrl-Clicking a ffi extern "C" function (neither to the .rs file defining the extern "C" method nor to the actual c/c++ code (header files are in path).

The crate in question uses a dynamic build script that either generates the bindings or uses pre-generated ones. in VS-Code with rust-analyzer I had to enable `rust-analyzer.procMacro.enable` for it to recognize this import.

## Steps to reproduce
Cargo.toml
```toml
[dependencies]
libmpv-sys = { version = "3.1.0" }
```
main.rs
```rust
use libmpv_sys::mpv_render_context_create;
```
Ctrl-Click on `mpv_render_context_create`

The line that includes the bindings:
https://github.com/ParadoxSpiral/libmpv-rs/blob/661518ddef152d8932ee22b1b68591182183efec/libmpv-sys/src/lib.rs#L23

I'm using rust via msys2. the following packages are installed:
mingw-w64-x86_64-rust
mingw-w64-x86_64-mpv

the `mpv_render_context_create` function is defined in `/mingw64/include/mpv/render.h`

Clion's rust is configured as following:
Toolchain location: `C:\msys64\mingw64\bin`
Standard library: `C:\msys64\mingw64\lib\rustlib\src\rust`
Clion's toolchains are configured as follows:
Environment: `C:\msys64\mingw64`
Cmake: `C:\msys64\mingw64\bin\cmake`
make: `C:\msys64\mingw64\bin\make`
C Compiler: `C:\msys64\mingw64\bin\gcc`
C++ Compiler: `C:\msys64\mingw64\bin\g++`
Debugger `C:\msys64\mingw64\bin\gdb`

**Comment by Undin** [comment]
@zumoshi try to enable `org.rust.cargo.evaluate.build.scripts` experimental feature via the corresponding dialog. To open the dialog, just invoke `Help | Find Action` action and type `Experimental features`

**Comment by zumoshi** [comment]
Sorry for missing that option before opening the issue. with `org.rust.cargo.evaluate.build.scripts` ctrl-clicking displays the rust file where the `extern "C"` method was defined.
It would be nice if ctrl-clicking on the definition would go to the `.h` file for the original C method. but not a requirement.
Thanks.

**Comment by Undin** [comment]
> It would be nice if ctrl-clicking on the definition would go to the .h file for the original C method. but not a requirement.

Yep, it would be great to have navigation to C functions from extern Rust functions. But I think it should be implemented in a similar way as it currently implemented for Rust/JS interop in #5933

**Comment by Undin** [comment]
I've created a separate issue for Rust/C navigation - #6639
